### ISSUE TITLE
Create analysis and recommendations report based on initial model performance

### ISSUE THREAD

**Issue by bamader** [CDCgov/dibbs-text-to-code#132]
Now that we have our full collection of preliminary model performance data, we need to analyze the performance metrics we've collected. We want to do this with an eye towards developing a sound proposal to take forward into the MVP phase when we begin fine-tuning and additional training. The scope of this ticket is to write a document capturing the details of what we've done to evaluate the Proof of Concept phase. Specifically, we want to include sections like:

* Abstract and executive summary (because the takeaways at a glance are important for any longer report)
* Introduction and problem statement--what text to code is trying to do as a whole is clear, but what isn't clear is why we need to evaluate 21 different models off the shelf before we do anything. Walk the reader through why we're doing this pre-work and why it's so important (e.g. model considerations like size in number of parameters, search accuracy, encoding speed, and memory footprint; training considerations like choice of algorithm, optimization function, length in epochs or iterations, supervision vs unsupervised, and nature of the training data; determining whether it's worth pursuing clinical models or general purpose and then domain + task adapting; investigating whether small vs large models learn patterns better vs memorizing data examples; determining whether this approach is even feasible and getting an accuracy baseline; identifying whether we need a re-ranker; etc.)
* Testing framework and methodology: summarize how we built our synthetic validation set (including all those nifty data enhancements we did) and how we selected the final 60k for evaluation; discuss any considerations of the Azure testing environment such as impact of GPU
* Describe the stats we measured and what each of them means in context
* Link to the data table (it's too large to fit cleanly on a page in the report)
* Observations: highlight key trends and takeaways present in the data that are worth noting
* Analysis: use the observations and data to answer the problem statement questions directly and summarize these into concrete actionable insights
* Recommendations: use the previous analysis to drive recommendations for how to proceed with a subset of models in the fine-tuning phase
